buildtest/utils/file.py

```diff
@@ -143,3 +143,80 @@ def resolve_path(path):
 
     if os.path.exists(real_path):
         return real_path
+
+def read_file(filepath):
+    """ This method provides capability to read a file.
+
+        Parameters:
+
+        :param filepath: file name to read
+        :type filepath: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath,str):
+        return None
+
+    # resolve_path will handle shell and user expansion and account for any symlinks and check for file existence.
+    # if resolve_path does not return gracefully it implies file does not exist and will return None
+    filepath = resolve_path(filepath)
+
+    # if it's invalid file let's return None
+    if not filepath:
+        return None
+
+    content = []
+    fd = open(filepath, "r")
+
+    while True:
+
+        line = fd.readline()
+        if line:
+            content.append(line)
+        else:
+            break
+
+    return content
+
+def write_file(filepath, content):
+    """ This method provides capability to write a file.
+
+        Parameters:
+
+        :param filepath: file name to write
+        :type filepath: str, required
+        :param content: content to write to file
+        :type content: list, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath,str):
+        return None
+
+    # if content was passed as a string, let's convert to list
+    if isinstance(content, str):
```

Comment:
This also doesn't make sense - just take the content given from the user, and check that it's a string, and write it to the file. If the user wants newlines, etc., this is up for him/her to pass.

Reply:
just to clarify we should only accept ``str`` type to ``write_file`` when passing the content of file. If it's not a string we can return ``None``

Reply:
I think it's fine to assume that a string is passed, and let the error be raised if something else is. I would just check that the filename exists for read, and that the dirname exists for write.

Reply:
So i initially allowed an input of list because ``BuildTestCommand.execute`` returns output, error as a ``list``. These are then invoked to ``write_file``.
Just to confirm you would want input content to be in ``string``, if it's list would you want ``write_test`` to do nothing or convert to string and try to write file.

Reply:
`write_file("\n".join(output))`

Reply:
Unrelated to this, but i was testing locally something is wrong with ``BuildTestCommand`` it was working before now it's only getting first line of file and i can reproduce this on the client. The input content ``out`` and ``err`` are invoked via ``write_file`` so i was trying to figure out if this was an issue with my function, seems like it's coming from ``BuildTestCommand``

```
>>> from buildtest.utils.command import BuildTestCommand
>>> cmd = BuildTestCommand("python .buildtest/python-circle/circle_area.py")
>>> out,err = cmd.execute()
>>> out
['Circle Radius  2\n']

>>> os.system("python .buildtest/python-circle/circle_area.py")
Circle Radius  2
Area of circle  12.566370614359172
Block
0

```

Reply:
This issue is present in ``devel`` branch, ill open an issue basically its reading first line and stopping.